sycl/doc/extensions/experimental/sycl_ext_oneapi_complex.asciidoc

```diff
@@ -211,17 +221,125 @@ namespace sycl::ext::oneapi::experimental {
 } // namespace sycl::ext::oneapi::experimental
 ```
 
-=== Mathematical operations
+=== Marray Complex Class Specialization
+
+This proposal also introduces the specialization of the marray class to
+support SYCL complex. The marray class undergoes slight modification for this
+specialization, primarily involving the removal of operators that are
+inapplicable. No new functions or operators are introduced to the marray class.
+
+The marray complex specialization maintains the principles of trivial
+copyability (as seen in the Complex class description), with the
+`is_device_copyable` type trait resolving to `std::true_type`.
+
+The marray definition used within this proposal assumes that any operator the
+`sycl::marray` class defines is only implemented if the marray's value type
+also implements the operator.
+
+For instance,
+`sycl::marray<sycl::ext::oneapi::experimental::complex<T>, NumElements>` does
+not implement the modulus operator since
+`sycl::ext::oneapi::experimental::complex<T>` does not support it.
```

Comment:
I think you should replace this block with comments on the operators themselves.

SYCL does this by adding an "Available only when" comment, see [here](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_math_array_interface) for an example.

Reply:
I understand your suggestion, however, this is not what this "example" is trying to show.

What this example is trying to show is that the `marray` specialization only implements the operator that `complex` implements.
They will never be "Available in certain condition" since the `marray` is specialized for `complex`.
That's why the structure shown after this block only shows the operator deleted.

This might be clearer:

```
-The marray definition used within this proposal assumes that any operator the
-`sycl::marray` class defines is only implemented if the marray's value type
-also implements the operator.
+The `marray` specialization used within this proposal assumes that any operator
+the `sycl::marray` class defines is only implemented if the `marray`'s value
+type also implements the operator.
 
-For instance,
-`sycl::marray<sycl::ext::oneapi::experimental::complex<T>, NumElements>` does
-not implement the modulus operator since
-`sycl::ext::oneapi::experimental::complex<T>` does not support it.
+Because the `marray` is specialized for `complex`; as an example, the modulus
+operator is not implemented since `sycl::ext::oneapi::experimental::complex<T>`
+does not support it.
```

Reply:
Oh, I see.

You're right; if you're using a specialization, then we shouldn't use "Available only when" for this.  I think what threw me off was your use of the word "assumes" -- I read that as something like "This proposal is relying on behavior not documented here", which made me go and look at how `sycl::marray` was defined in the SYCL spec.

I would recommend something even simpler than what you suggested.  It seems to me that something like the below would be sufficient to document the class and summarize what the class synopsis is trying to convey:

> The `marray` specialization for `complex<T>` deletes any operator that is not supported by `complex<T>`.